src/main/java/se/iths/SpringbootGroupProject/controllers/GuestController.java

```diff
@@ -0,0 +1,30 @@
+package se.iths.SpringbootGroupProject.controllers;
+
+import org.springframework.web.bind.annotation.*;
+import se.iths.SpringbootGroupProject.dto.PublicMessageAndUsername;
+import se.iths.SpringbootGroupProject.repositories.MessageRepository;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/")
+public class GuestController {
+
+    private final MessageRepository messageRepository;
+
+    public GuestController(MessageRepository messageRepository) {
+        this.messageRepository = messageRepository;
+    }
+
+    @GetMapping("messages")
+    List<PublicMessageAndUsername> all(){
+            return messageRepository.findAllByPrivateMessageIsFalse()
+                    .stream()
+                    .map(message -> new PublicMessageAndUsername(
+                            message.getDate(),
+                            message.getTitle(),
+                            message.getMessageBody(),
+                            message.getUser().getUserName())).toList();
+    }
```

Comment:
Should we also have a @GetMapping("users") for Users to get a list of usernames? Or if this is the only method in GuestController maybe we can remove the endpoint "messages" and just get this information from /api/

Reply:
> Should we also have a @GetMapping("users") for Users to get a list of usernames? Or if this is the only method in GuestController maybe we can remove the endpoint "messages" and just get this information from /api/

We could add an endpoint to "users". At the moment the only public data we know of is from the public messages, but I do think the amount of endpoints will add on over time, hence the specific mapping of messages.

Reply:
Okay nice, then I think we can accept this and merge it into our main. As with everything else in our project, it can always be added to later if needed 😊 Good work!